src/Message/Reducer.php

```diff
@@ -93,16 +97,32 @@ public function finalize(Closure $closure): self
         return $this;
     }
 
+    /** @param iterable<Message> $messages */
+    public function messages(iterable $messages): self
+    {
+        $this->messages = $messages;
+
+        return $this;
+    }
+
     /**
-     * @param iterable<Message> $messages
+     * @param iterable<Message>|null $messages
      *
      * @return OUT|STATE
      * @psalm-return (OUT is STATE ? STATE : OUT)
      */
-    public function reduce(iterable $messages): array
+    public function reduce(iterable|null $messages = null): array
     {
         $state = $this->initState;
 
+        if ($messages === null) {
```

Comment:
Why are we changing the API how to provide the messages? I did not see any usage here for the extra `Reducer::messages` method. This seems to just complicate things, no?


In this example, `messagesA` is unused, this could lead to confusion.
```php
$reducer = (new Reducer)->messages($messagesA);

$reducers->reduce($messagesB);
```